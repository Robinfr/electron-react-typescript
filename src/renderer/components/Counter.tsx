import * as React from 'react';

require('./Counter.scss');
const redCubeImg = require('./RedCube.jpg');

export interface Props {
    value: number;

    incrementValue: () => any;
    decrementValue: () => any;
}

const Counter: React.FunctionComponent<Props> = ({ value, incrementValue, decrementValue }) => (
    <div className="counter">
        <p>
            <img src={redCubeImg} />
        </p>
        <p id="counter-value">Current value: {value}</p>
        <p>
            <button id="increment" onClick={incrementValue}>
                Increment
            </button>
            <button id="decrement" onClick={decrementValue}>
                Decrement
            </button>
        </p>
    </div>
);

export default Counter;
