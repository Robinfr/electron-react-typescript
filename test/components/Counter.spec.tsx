import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Counter from '../../src/renderer/components/Counter';

describe('Counter component', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Counter value={1} incrementValue={jest.fn()} decrementValue={jest.fn()} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
