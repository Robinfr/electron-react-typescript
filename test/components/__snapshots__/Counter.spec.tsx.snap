// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component renders correctly 1`] = `
<div
  className="counter"
>
  <p>
    <img
      alt="Red cube"
      src="test-file-stub"
    />
  </p>
  <p
    id="counter-value"
  >
    Current value: 
    1
  </p>
  <p>
    <button
      id="increment"
      onClick={[MockFunction]}
    >
      Increment
    </button>
    <button
      id="decrement"
      onClick={[MockFunction]}
    >
      Decrement
    </button>
  </p>
</div>
`;
